persist/fs/mmap_linux.go

```diff
@@ -18,17 +18,50 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package commitlog
-
-import "testing"
-
-// go test -bench=LemireCreate
-// see http://lemire.me/blog/2016/09/22/swift-versus-java-the-bitset-performance-test/
-func BenchmarkLemireCreate(b *testing.B) {
-	for i := 0; i < b.N; i++ {
-		bitmap := newBitSet(0) // we force dynamic memory allocation
-		for v := uint(0); v <= 100000000; v += 100 {
-			bitmap.set(v)
-		}
+package fs
+
+import "syscall"
+
+const (
+	hugePageSize = 2 << 20 // 2mb
+)
+
+func mmap(fd, offset, length int, opts mmapOptions) ([]byte, error) {
+	if length == 0 {
+		// Return an empty slice (but not nil so callers who
+		// use nil to mean something special like not initialized
+		// get back an actual ref)
+		return make([]byte, 0), nil
+	}
+
+	var prot int
+	if opts.read {
+		prot = prot | syscall.PROT_READ
+	}
+
+	flags := syscall.MAP_SHARED
+	b, err := syscall.Mmap(fd, int64(offset), length, prot, flags)
+	if err != nil {
+		return nil, err
+	}
+
+	if length < hugePageSize {
```

Comment:
Not sure I follow, why not use a huge page for 1.5MB index file?

Reply:
Well we definitely don't want to use huge pages for anything less than 4096 or even "a few" 4096 files in length, otherwise you do cause memory fragmentation (it's a problem with huge pages).  Maybe I'll use > 32kb which is 8 pages.